core/frontend/src/test/tile/IndexedDBCache.test.ts

```diff
@@ -0,0 +1,126 @@
+    return;
+  }
+
+  public async testAddContent(uniqueId: string, content: ArrayBuffer) {
+    await this.addContent(uniqueId, content);
+  }
+
+  public async testRetrieveContent(uniqueId: string): Promise<ArrayBuffer | undefined>{
+    return this.retrieveContent(uniqueId);
+  }
+
+  public async testClose() {
+    await this.close();
+    return;
+  }
+
+  public async convertStringToBuffer(str: string): Promise<ArrayBuffer> {
+    return new TextEncoder().encode(str).buffer;
+  }
+
+  public async convertBufferToString(buffer: ArrayBuffer): Promise<string> {
+    return new TextDecoder().decode(new Uint8Array(buffer)).toString();
+  }
+}
+
+describe("IndexedDBCache", () => {
+  const cache = new TestCache("TestDB");
+
+  it("should add content to the cache", async () => {
+    const uniqueId = "test-add";
+    const content = await cache.convertStringToBuffer(uniqueId);
+    await cache.testAddContent(uniqueId, content);
+
+    const retrievedContent = await cache.testRetrieveContent(uniqueId);
+    const strRetrievedContent = await cache.convertBufferToString(retrievedContent!);
+    expect(strRetrievedContent).equal(uniqueId);
+  });
+
+  it("should delete content from the cache", async () => {
+    const uniqueId = "test-delete";
+    const content = await cache.convertStringToBuffer(uniqueId);
+
+    await cache.testAddContent(uniqueId, content);
+    await cache.testDeleteContent(uniqueId);
+
+    const retrievedContent = await cache.testRetrieveContent(uniqueId);
+    expect(retrievedContent).equal(undefined);
+  });
+
+  it("should fetch content from the cache if available", async () => {
+    const uniqueId = "test-fetch-cache";
+    const content = await cache.convertStringToBuffer(uniqueId);
+
+    await cache.testAddContent(uniqueId, content);
+
+    const fetchedContent = await cache.fetch(uniqueId, async () => {
+      throw new Error("This callback should not be called");
+    });
+    const strFetchedContent = await cache.convertBufferToString(fetchedContent);
+    expect(strFetchedContent).equal(uniqueId);
+  });
+
+  it("should fetch content from the network if not available in the cache", async () => {
+    const uniqueId = "test-fetch-network";
+    const content = await cache.convertStringToBuffer(uniqueId);
+
+    const fetchedContent = await cache.fetch(uniqueId, async () => {
+      return new Response(content);
+    });
+    const strFetchedContent = await cache.convertBufferToString(fetchedContent);
+    expect(strFetchedContent).equal(uniqueId);
+  });
+
+  it("should fetch content from the network if expired in the cache", async () => {
+    const pause = async (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+    const expiringCache = new TestCache("ExpiringTestDB", 1);
+    const uniqueId = "test-expiring";
+    const content = await cache.convertStringToBuffer(uniqueId);
+
+    await expiringCache.testAddContent(uniqueId, content);
+    await pause(1000);
```

Comment:
Do you mean because of the pause?

Reply:
I changed the test to now have an expiration time of -1 so that it expires immediately with no reliance on a pause. I changed the test where the cache does not expire to have a much larger expiration time to avoid flakiness in that test as well.